.infrastructure/helm-chart/todoapp/charts/mysql/templates/statefulSet.yml

```diff
@@ -0,0 +1,89 @@
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ .Chart.Name }}
+  namespace: {{ .Values.mysql.namespace}}
+spec:
+  selector:
+    matchLabels:
+      app: {{ .Chart.Name }}
+  replicas: {{ .Values.mysql.replicas }}
+  serviceName: {{ .Chart.Name }}
+  template:
+    metadata:
+      labels:
+        app: {{ .Chart.Name }}
+    spec:
+      containers:
+      - name: {{ .Chart.Name }}
+        image: {{ .Values.mysql.image.repository }}:{{ .Values.mysql.image.tag }}
+        env:
+        {{- range $k, $v := .Values.mysql.secret }}
+        - name: {{ $k }}
+          valueFrom:
+            secretKeyRef:
+              name: mysql-secrets
+              key: {{ $k }}
+        {{- end }}
```

Comment:
Ensure that the keys in `.Values.mysql.secret` are correctly defined in your `values.yaml` file. These keys should match the secret keys you intend to use.